src/components/chips/js/chipsController.js

```diff
@@ -0,0 +1,86 @@
+(function () {
+  'use strict';
+  angular
+      .module('material.components.chips')
+      .controller('MdChipsCtrl', MdChipsCtrl);
+
+  function MdChipsCtrl ($scope, $element, $q, $mdUtil, $mdConstant, $log) {
+    //-- private variables
+    var self = this;
+    this.$mdConstant = $mdConstant;
+
+    /** @type {!angular.NgModelController} */
+    this.ngModelCtrl = null;
+
+    /** @type {Array.<Object>} */
+    this.items = [];
+
+    //-- public variables
+    self.scope = $scope;
+
+    return self.init();
+
+    // How to update the model:
+    // self.ngModelCtrl.$setViewValue(new Date(Number(this.dataset.timestamp)));
+    // self.ngModelCtrl.$render();
+
+  }
+
+  /** 
+   * The model used by the input element.
+   * @type {!String}
+   */
+  MdChipsCtrl.prototype.itemBuffer = '';
```

Comment:
can confirm - checked it out in the JSCompiler styleguide... Interesting...
Social team should probably be told this. We did this all over the place in
Glass, after stealing some of their engineers and basing our entire stack
off theirs...

On Tue, Mar 24, 2015 at 10:46 AM Jeremy Elbourn <notifications@github.com>
wrote:

> In src/components/chips/js/chipsController.js
> <https://github.com/angular/material/pull/1981#discussion_r27053882>:
>
> > +    //-- public variables
> > +    self.scope = $scope;
> > +
> > +    return self.init();
> > +
> > +    // How to update the model:
> > +    // self.ngModelCtrl.$setViewValue(new Date(Number(this.dataset.timestamp)));
> > +    // self.ngModelCtrl.$render();
> > +
> > +  }
> > +
> > +  /**
> > +   * The model used by the input element.
> > +   * @type {!String}
> > +   */
> > +  MdChipsCtrl.prototype.itemBuffer = '';
>
> Even in Closure-land, instance properties don't go on the prototype- only
> methods.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/angular/material/pull/1981/files#r27053882>.
>